LogInMekashron/LogInMekashron/Services/LogIn/LogInService.cs

```diff
@@ -13,23 +13,24 @@ public class LogInService : ILoginService
     {
         private const string Url = "http://isapi.mekashron.com/StartAJob/General.dll/soap/ILogin";
         private IRestService _restService;
+        public XDocument doc { get; set; }
 
         public LogInService()
         {
             _restService = new RestService();
         }
 
-        public async Task Login(string login, string password)
+        public async Task<XDocument> Login(string login, string password)
         {
             var soapString = this.ConstructSoapRequest(login, password, ipaddress);
             var content = new StringContent(soapString, Encoding.UTF8, "application/xml");
-            await _restService.GetAsync<string>(Url, content);
-            return;
+            var doc = await _restService.GetAsync<string>(Url, content);
+            return doc;
         }
 
         string ipaddress = DependencyService.Get<IIPAddressManager>().GetIPAddress();
```

Comment:
You should create variable of service type and separate resolving service instance and getting IP address